packages/web-components/src/calendar/calendar.ts

```diff
@@ -0,0 +1,368 @@
+   */
+  @attr public weekdayFormat: WeekdayFormat = WeekdayFormat.narrow;
+
+  /**
+   * the year on the month picker
+   */
+  @attr public monthPickerYear: number = new Date().getFullYear();
+
+  /**
+   * the decade on the year picker
+   */
+  @attr public yearPickerDecade: number = this.monthPickerYear - (this.monthPickerYear % 10);
+
+  /**
+   * whether the year picker is open
+   */
+  @attr public yearPickerOpen: boolean = false;
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    this.addEventListener('dateselected', this.dateSelectedHandler);
+    this.addEventListener('rightcellselected', this.rightCellSelectedHandler);
+  }
+
+  public disconnectedCallback() {
+    this.removeEventListener('dateselected', this.dateSelectedHandler);
+    this.removeEventListener('rightcellselected', this.rightCellSelectedHandler);
+    super.disconnectedCallback();
+  }
+
+  /**
+   * Gets data needed to render about a month picker year as well as the previous and next years
+   * @param year - year of the month picker
+   * @returns - an object with data about the current and 2 surrounding years
+   * @public
+   */
+  public getMonthPickerInfo(year: number = this.monthPickerYear): MonthPickerInfo {
+    return {
+      year,
+      previous: year - 1,
+      next: year + 1,
+    };
+  }
+
+  /**
+   * Gets data needed to render about a month picker year as well as the previous and next years
+   * @param decadeStart - the start of the decade on the year picker
+   * @returns - an object with data about the current and 2 surrounding decades
+   * @public
+   */
+  public getYearPickerInfo(decadeStart: number = this.yearPickerDecade): YearPickerInfo {
+    return {
+      decadeStart,
+      decadeEnd: decadeStart + 11,
+      previousStart: decadeStart - 12,
+      nextStart: decadeStart + 12,
+    };
+  }
+
+  /**
+   * Changes the month and year on the calendar
+   * @param month - the month to be switched to
+   * @param year - the year to be switched to
+   */
+  public switchMonth(month: number, year: number) {
+    this.selectedDates = '';
+
+    this.year = year;
+    this.month = month;
+
+    this.monthPickerYear = year;
+    this.yearPickerDecade = year - (year % 10);
+  }
+
+  /**
+   * Open/close the year picker
+   * @public
+   */
+  public toggleYearPicker() {
+    this.yearPickerOpen = !this.yearPickerOpen;
+    this.yearPickerDecade = this.monthPickerYear - (this.monthPickerYear % 10);
+  }
+
+  /**
+   * Creates a class string for the "Go to today" link
+   * @returns - string of class names
+   * @public
+   */
+  public getLinkClassNames() {
+    const today: Date = new Date();
+
+    //when the month picker is not visible, this.monthPickerYear is always going to be the current year.
+    return this.month === today.getMonth() + 1 &&
+      this.year === today.getFullYear() &&
+      this.monthPickerYear === today.getFullYear()
+      ? 'slotted-link inactive'
+      : 'slotted-link';
+  }
```

Comment:
We might want to abstract this logic out into a separate helper method of `isToday()` that returns if the day is today. You could then simplify the ternary logic here to: `return this.isToday() ? 'slotted-link inactive' : 'slotted-link';`

Reply:
I decided to remove this method since it's just one line, and just applied the logic in my template.
```typescript
class="${x => x.isToday()? 'slotted-link inactive' : 'slotted-link'}"
```